File: Práctica 1/S1/src/s1/factoriaCarretera.java
package s1;

import java.util.ArrayList;

public class factoriaCarretera extends factoriaCarrerayBicicleta {

	public carreraCarretera crearCarrera(int codigo) {
		carreraCarretera carrera = new carreraCarretera(codigo);
                return carrera;
	}

	public bicicletaCarretera crearBicicleta(int id) {
		bicicletaCarretera bicicleta = new bicicletaCarretera(id);
                return bicicleta;
	}
}

File: Práctica 1/S1R/lib/FactoriaCarretera.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module Ruby
  class FactoriaCarretera < FactoriaCarrerayBicicleta
    def crearCarrera()
      carrera = CarreraCarretera.new.clonar
      return carrera
    end
    def crearBicicleta()
      bicicleta = BicicletaCarretera.new.clonar
      return bicicleta
    end
  end
end
